mubench.pipeline/data/detector.py

```diff
@@ -11,24 +11,27 @@ class Detector:
     BASE_URL = "http://www.st.informatik.tu-darmstadt.de/artifacts/mubench/detectors"
     RELEASES_FILE = "releases.yml"
 
-    def __init__(self, detectors_path: str, detector_id: str, java_options: List[str]):
+    def __init__(self, detectors_path: str, detector_id: str, java_options: List[str], requested_release: Optional[str] = None):
         self.id = detector_id
         self.base_name = detector_id.split("_", 1)[0]
         self.path = join(detectors_path, self.id)
 
         releases_index_path = join(self.path, Detector.RELEASES_FILE)
-        release_info = self._get_release_info(releases_index_path)
-        self.release_tag = release_info.get("tag", "latest")
+        release_info = self._get_release_info(releases_index_path, requested_release)
+        release_tag = release_info.get("tag", "latest")
+        cli_version = release_info.get("cli_version", None)
         self.md5 = release_info.get("md5", None)
-        self.cli_version = release_info.get("cli_version", None)
 
         self.jar_path = join(self.path, self.base_name + ".jar")
-        self.jar_url = "{}/{}/{}/{}.jar".format(Detector.BASE_URL, self.release_tag, self.cli_version, self.base_name)
-        self.runner_interface = RunnerInterface.get(self.cli_version, self.jar_path, java_options)
+        self.jar_url = "{}/{}/{}/{}.jar".format(Detector.BASE_URL, release_tag, cli_version, self.base_name)
 
-    def _get_release_info(self, releases_index_file_path: str) -> Dict[str, str]:
+        self.runner_interface = RunnerInterface.get(cli_version, self.jar_path, java_options)
+
+    def _get_release_info(self, releases_index_file_path: str, requested_release: Optional[str]) -> Dict[str, str]:
         if exists(releases_index_file_path):
             releases = read_yaml(releases_index_file_path)
+            if requested_release:
+                releases = [r for r in releases if r.get("tag", "latest") == requested_release]
             if len(releases) > 0:
```

Comment:
Agreed. We should raise a ValueError to indicate that the tag did not exist. Theoretically, I think we could load the detectors on startup and use the releases info to validate the CLI arguments, but I wouldn't go this far at this point.